### DISCUSSION TITLE
E-mail address in LDAP response?

### DISCUSSION THREAD

**Discussion by jchonig** [kanidm/kanidm#3189] [Q&A] [answered]
I set the E-mail address on a person account and it shows up in `kanidm person get` and in the Web GUI.

But I don't see it returned with an of a person.  Am I missing something or is this an omission?

**Comment by Firstyear** [comment, marked as answer]
email is PII and is not accessible over ldap by default. See https://kanidm.github.io/kanidm/stable/integrations/ldap.html#access-controls

If you want to read the email address you need to use a service-account https://kanidm.github.io/kanidm/stable/accounts/service_accounts.html with an api token and the appropriate read permissions.  https://kanidm.github.io/kanidm/stable/access_control/intro.html#default-permission-groups

**Comment by jchonig** [comment]
Thanks for the quick answer.  I am using a service account for real use, just testing with ldapsearch.

Why is an `mail` PII, but not `displayname`?  Just curious.

Thanks.

Jeff

**Reply by Firstyear** [reply]
displayname is what the person chooses to be seen as.

*legal name* is the true legal name of the person, that is pii.
